### ISSUE TITLE
inline-scan:latest Broken Image

### ISSUE THREAD

**Issue by leeman24** [anchore/ci-tools#13]
First time using this image but found https://anchore.freshdesk.com/support/solutions/articles/36000058199-gitlab-integration and using the docker.io/anchore/inline-scan:latest image is broken. I was able to run it by using an older tag docker.io/anchore/inline-scan:v0.3.4.

Looks like the `/docker-entrypoint.sh` doesn't exist:

```
$ anchore-cli system wait
/usr/local/bin/anchore-cli: line 3: /docker-entrypoint.sh: No such file or directory
```

```
Running with gitlab-runner 12.0.2 (d0b76032)
  on docker-auto-scale ed2dce3a
Using Docker executor with image docker.io/anchore/inline-scan:latest ...
Starting service docker.io/anchore/inline-scan:latest ...
Pulling docker image docker.io/anchore/inline-scan:latest ...
Using docker image sha256:fc0e5d89a4e419e2dd6511122aec6741f661f4f635d0d3b96ae81c1c49970946 for docker.io/anchore/inline-scan:latest ...
Waiting for services to be up and running...

*** WARNING: Service runner-ed2dce3a-project-13065579-concurrent-0-docker.io__anchore__inline-scan-0 probably didn't start properly.

Health check error:
service "runner-ed2dce3a-project-13065579-concurrent-0-docker.io__anchore__inline-scan-0-wait-for-service" timeout

Health check container logs:


Service container logs:
2019-07-08T18:01:26.216621216Z Starting Postgresql... Postgresql started successfully!
2019-07-08T18:01:29.354938684Z Starting Docker registry... Docker registry started successfully!
2019-07-08T18:01:29.355065194Z Starting Anchore Engine...

*********

Running on runner-ed2dce3a-project-13065579-concurrent-0 via runner-ed2dce3a-srm-1562608667-60e73e90...
Skipping Git repository setup
Skipping Git checkout
Skipping Git submodules setup
Downloading artifacts for container_build (247493437)...
Downloading artifacts from coordinator... ok        id=247493437 responseStatus=200 OK token=Q3zDfJnK
$ anchore-cli system wait
/usr/local/bin/anchore-cli: line 3: /docker-entrypoint.sh: No such file or directory
ERROR: Job failed: exit code 1
```

```
Running with gitlab-runner 12.0.2 (d0b76032)
  on docker-auto-scale 0277ea0f
Using Docker executor with image docker.io/anchore/inline-scan:v0.3.4 ...
Starting service docker.io/anchore/inline-scan:v0.3.4 ...
Pulling docker image docker.io/anchore/inline-scan:v0.3.4 ...
Using docker image sha256:63ee3fbfe207a726f870efcc124d70c0466542724bc2853abd8e7133b4609b8e for docker.io/anchore/inline-scan:v0.3.4 ...
Waiting for services to be up and running...
Running on runner-0277ea0f-project-13065579-concurrent-0 via runner-0277ea0f-srm-1562609402-97255c0d...
Skipping Git repository setup
Skipping Git checkout
Skipping Git submodules setup
Downloading artifacts for container_build (247505785)...
Downloading artifacts from coordinator... ok        id=247505785 responseStatus=200 OK token=FpCmNmkZ
$ anchore-cli system wait
Starting checks to wait for anchore-engine to be available timeout=-1.0 interval=5.0
API availability: Checking anchore-engine URL (http://anchore-engine:8228/v1)...
API availability: Success.
Service availability: Checking for service set (catalog,apiext,policy_engine,simplequeue,analyzer)...
Service availability: Checking for service set (catalog,apiext,policy_engine,simplequeue,analyzer)...
Service availability: Checking for service set (catalog,apiext,policy_engine,simplequeue,analyzer)...
Service availability: Success.
Feed sync: Checking sync completion for feed set (vulnerabilities)...
Feed sync: Success.
$ anchore-cli registry add "$CI_REGISTRY" gitlab-ci-token "$CI_JOB_TOKEN" --skip-validate
Registry: registry.gitlab.com
User: gitlab-ci-token
Type: docker_v2
Verify TLS: True
Created: 2019-07-08T18:14:13Z
Updated: 2019-07-08T18:14:13Z


$ anchore_ci_tools.py -a -r --timeout 500 --image $IMAGE_TAG

Image submitted to Anchore Engine: mysql:devops-pipeline-build-image
Waiting for analysis to complete...

	Status: not_analyzed
	Status: analyzing............
	Status: analyzed

Analysis completed!

Successfully generated anchore-reports/mysql_devops-pipeline-build-image-content-os.json.
Successfully generated anchore-reports/mysql_devops-pipeline-build-image-content-files.json.
Successfully generated anchore-reports/mysql_devops-pipeline-build-image-vuln.json.
Successfully generated anchore-reports/mysql_devops-pipeline-build-image-details.json.
Successfully generated anchore-reports/mysql_devops-pipeline-build-image-policy.json.
Uploading artifacts...
anchore-reports/*: found 5 matching files          
Uploading artifacts to coordinator... ok            id=247505786 responseStatus=201 Created token=LYyXejzs
Job succeeded
```

**Comment by Btodhunter** [comment]
Thanks for the report @leeman24! The latest version of the inline-scan container (v0.4.1) no longer has this issue.

I'm guessing that your using the inline-scan container as a service, like the example on our GitLab account here - https://gitlab.com/anchore/gitlab-demo/blob/master/.gitlab-ci.yml#L45-64

That example is more of an experimentation, and not really an "officially supported" method for running Anchore inline-scan. It only works on GitLab and is pretty finicky. If you're trying to run a stateless anchore-engine, the best method is to use the inline_scan script, which is detailed in this doc - https://docs.anchore.com/current/docs/engine/usage/integration/ci_cd/inline_scanning/

**Comment by leeman24** [comment]
Thanks @Btodhunter!

I was using the inline-scan container as a service as you mentioned. Good to know it is not officially supported. This work was mostly for a POC but I see the benefit of running this scan (even to just report issues and not fail the build). I will take a look at https://docs.anchore.com/current/docs/engine/usage/integration/ci_cd/inline_scanning/ if we choose to implement Anchore to our pipelines.

**Comment by Btodhunter** [comment]
I found that using the inline-scan image as a service often performs a little bit better than using the inline_scan script, however it's much less straight-forward and requires a bit more effort for the user. Therefor we're not recommending it for most users.

You can also run the open source anchore-engine on premises which will perform better than any integrated CI options. The on-prem solution provides persistence for ongoing image analysis, reporting and auditing purposes. The on-prem solution also has an enterprise component available that offers a UI, a reporting API, RBAC & an on-prem feeds service.

Thanks for giving Anchore a try in your pipelines! If you need any help with custom policies or getting your CI integration fleshed out, feel free to stop by our [community slack channel](https://anchore.com/slack)

**Comment by Btodhunter** [comment]
closed with 52c8fe5
